File: day20/.ts
function dropGifts(warehouse: string[][], drops: number[]): string[][] {
  let newWareHouse = warehouse.map(row => [...row])
  const rows = warehouse.length
  
  for (const col of drops){
    for (let row = rows - 1; row >= 0; row--){
      if (newWareHouse[row][col] == "."){
        newWareHouse[row][col] = "#"
        break
      }
    }
  }

  return newWareHouse
}

File: day20/.py
def drop_gifts(warehouse: list[list[str]], drops: list[int]) -> list[list[str]]:
  newWareHouse = [row[:] for row in warehouse]
  rows = len(warehouse)
  
  for col in drops:
    for row in range(rows - 1, -1, -1):
      if newWareHouse[row][col] == ".":
        newWareHouse[row][col] = "#"
        break
  return newWareHouse
